plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/changedList/ChangedFolderNode.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.git.client.compare.changedList;
+
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.ide.api.project.node.AbstractTreeNode;
+import org.eclipse.che.ide.api.project.node.HasAction;
+import org.eclipse.che.ide.api.project.node.Node;
+import org.eclipse.che.ide.project.shared.NodesResources;
+import org.eclipse.che.ide.ui.smartTree.presentation.HasPresentation;
+import org.eclipse.che.ide.ui.smartTree.presentation.NodePresentation;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * Node Element used for setting it to TreeNodeStorage and viewing folders that contain changed files.
+ *
+ * @author Igor Vinokur
+ */
+public class ChangedFolderNode extends AbstractTreeNode implements HasPresentation, HasAction {
+
+    private String           name;
+    private NodePresentation nodePresentation;
+
+    private final NodesResources nodesResources;
+
+    /**
+     * Create instance of ChangedFolderNode.
+     *
+     * @param name
+     *         name of the folder that represents this node
+     * @param nodesResources
+     *         resources that contain icons
+     */
+    public ChangedFolderNode(String name, NodesResources nodesResources) {
+        this.name = name;
+        this.nodesResources = nodesResources;
+    }
+
+    @Override
+    protected Promise<List<Node>> getChildrenImpl() {
+        return this.getChildren(false);
```

Comment:
`getChildren` method in `AbstractTreeNode` has following code:

```
    public final Promise<List<Node>> getChildren(boolean forceUpdate) {
        if (children == null || children.isEmpty() || forceUpdate) {
            return getChildrenImpl().then(setParentAndSaveState());
        }

        return Promises.resolve(children);
    }
```
you may have cyclic calling.